src/main/java/gregtech/integration/exnihilo/ExNihiloModule.java

```diff
@@ -0,0 +1,164 @@
+
+import gregtech.api.GTValues;
+import gregtech.api.gui.resources.SteamTexture;
+import gregtech.api.modules.GregTechModule;
+import gregtech.api.recipes.RecipeMap;
+import gregtech.api.recipes.RecipeMapBuilder;
+import gregtech.api.recipes.builders.SimpleRecipeBuilder;
+import gregtech.api.unification.material.event.MaterialEvent;
+import gregtech.api.unification.material.info.MaterialIconType;
+import gregtech.api.unification.ore.OrePrefix;
+import gregtech.api.util.Mods;
+import gregtech.common.items.MetaItems;
+import gregtech.common.metatileentities.MetaTileEntities;
+import gregtech.integration.IntegrationModule;
+import gregtech.integration.IntegrationSubmodule;
+import gregtech.integration.exnihilo.items.ExNihiloPebble;
+import gregtech.integration.exnihilo.metatileentities.MetaTileEntitySieve;
+import gregtech.integration.exnihilo.metatileentities.MetaTileEntitySteamSieve;
+import gregtech.integration.exnihilo.recipes.ExNihiloRecipes;
+import gregtech.integration.exnihilo.recipes.SieveDrops;
+import gregtech.integration.exnihilo.recipes.ui.SieveUI;
+import gregtech.modules.GregTechModules;
+
+import net.minecraft.init.SoundEvents;
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.event.RegistryEvent;
+import net.minecraftforge.fml.common.event.FMLInitializationEvent;
+import net.minecraftforge.fml.common.event.FMLPreInitializationEvent;
+import net.minecraftforge.fml.common.eventhandler.EventPriority;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collections;
+import java.util.List;
+
+import static gregtech.api.unification.ore.OrePrefix.Conditions.hasOreProperty;
+import static gregtech.api.unification.ore.OrePrefix.Flags.ENABLE_UNIFICATION;
+import static gregtech.common.metatileentities.MetaTileEntities.*;
+
+@GregTechModule(
+                moduleID = GregTechModules.MODULE_EN,
+                containerID = GTValues.MODID,
+                modDependencies = Mods.Names.EX_NIHILO_CREATIO,
+                name = "GregTech Ex Nihilo Creatio Integration",
+                description = "Ex Nihilo Integration Module")
+public class ExNihiloModule extends IntegrationSubmodule {
+
+    // Items
+    public static ExNihiloPebble GTPebbles;
+
+    // Recipe maps
+    public static final RecipeMap<SimpleRecipeBuilder> SIEVE_RECIPES = new RecipeMapBuilder<>("electric_sieve",
+            new SimpleRecipeBuilder().duration(100).EUt(4))
+                    .itemInputs(2)
+                    .itemOutputs(36)
+                    .ui(SieveUI::new)
+                    .sound(SoundEvents.BLOCK_SAND_PLACE)
+                    .build();
+
+    // Machines
+    public static MetaTileEntitySteamSieve STEAM_SIEVE_BRONZE;
+    public static MetaTileEntitySteamSieve STEAM_SIEVE_STEEL;
+    public static MetaTileEntitySieve[] SIEVES = new MetaTileEntitySieve[GTValues.V.length - 1];
```

Comment:
Move all MTE and related registration code to a dedicated class. In the regular `MetaTileEntities` class add a comment in the appropriate spot stating the IDs used.